styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -152,7 +209,95 @@ public void poll(FlyteExecutionId flyteExecutionId, RunState runState)
     } catch (Exception e) {
       throw new PollingException(flyteExecutionId, e);
     }
+  }
+
+  void init() {
+    scheduleWithJitter(this::terminateDanglingFlyteExecutions, scheduledExecutor, terminateDanglingFlyteExecInterval);
+  }
+
+  @VisibleForTesting
+  void terminateDanglingFlyteExecutions() {
+    try {
+      // TODO: add tracing
+      tryTerminateDanglingFlyteExecutions();
+    } catch (Throwable t) {
+      LOG.warn("Error while terminating dangling flyte executions", t);
+    }
+
+  }
+
+  private void tryTerminateDanglingFlyteExecutions() {
+    var projects = flyteAdminClient.listProjects();
+
+    var allFutures = new ArrayList<CompletableFuture<Void>>();
+    for (var project : projects.getProjectsList()) {
+      for (var domain : project.getDomainsList()) {
+        String paginationToken = null;
+        do {
+          var executions =
+              flyteAdminClient.listExecutions(project.getId(), domain.getId(), 100, paginationToken, "");
+          executions.getExecutionsList().stream()
+              .filter(exec -> exec.getClosure().getPhase() == Execution.WorkflowExecution.Phase.RUNNING)
+              .map(exec -> AnnotatedFlyteExecutionId.create(
+                  FlyteExecutionId.fromProto(exec.getId()),
+                  exec.getSpec().getAnnotations().getValuesMap())
+              )
+              .filter(this::isFlyteExecutionTriggeredByStyx)
+              .map(annotatedId -> runAsync(guard(() -> tryTerminateDanglingFlyteExecution(annotatedId)), executor))
+              .forEach(allFutures::add);
+          paginationToken = executions.getToken();
+        } while (!"".equals(paginationToken));
+      }
+    }
+    allFutures.forEach(CompletableFuture::join);
+  }
+
+  private void tryTerminateDanglingFlyteExecution(AnnotatedFlyteExecutionId annotatedId) {
+    var workflowInstance = getWorkflowInstance(annotatedId);
+    var runState = stateManager.getActiveState(workflowInstance);
```

Comment:
are these operations  becoming a load concern?

Reply:
We are doing [the same](https://github.com/spotify/styx/blob/6a79ce45715076074123ca5662458077bbe6a939/styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java#L589) in `KubernetesDockerRunner`